### DISCUSSION TITLE
Change application command permissions

### DISCUSSION THREAD

**Discussion by dbodzioch** [Discord4J/Discord4J#1116] [Q&A] [answered]
I would like to change the command permissions. Can it be done from the client.getRestClient().getApplicationService()modifyApplicationCommandPermissions call with a token in the client from the admin panel? Or do I need another authorization?

**Comment by NovaFox161** [comment, marked as answer]
Application command permissions endpoints like the one you mentioned do not function. You will get an error attempting to use that method/endpoint.

Discord redid command permissions (Application Command Permissions V2) several months back. These changes require the bot to obtain an oauth token with the correct scope (`applications.commands.permissions.update`) included. 

Since Discord4J does not yet have Oauth2 support built in, this is impossible to do within the library as-is and requires additional scaffolding to be done on your side, or using an HTTP library to make these requests if you require them.
Here's Discord's docs on these endpoints: https://discord.com/developers/docs/interactions/application-commands#permissions
